src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -117,28 +124,37 @@ const SequenceBox = (props: SequenceBoxProps) => {
           <span className={styles.againstText}>Against</span>{' '}
           <span>{species.length} species</span>
         </div>
+        <div className={styles.showHideWrapper}>
+          <ShowHide
+            isExpanded={isExpanded}
+            onClick={() => setExpanded(!isExpanded)}
+          ></ShowHide>
+        </div>
       </div>
 
-      {blastResults.map((result) => {
-        const speciesInfo = species.filter(
-          (sp) => sp.genome_id === result.genomeId
-        );
-        return (
-          <SingleBlastJobResult
-            key={result.jobId}
-            species={speciesInfo[0]}
-            jobId={result.jobId}
-            diagramWidth={plotwidth}
-          />
-        );
-      })}
-      <div className={styles.resultsSummaryRow}>
+      {isExpanded &&
+        blastResults.map((result) => {
+          const speciesInfo = species.filter(
```

Comment:
Can't `find` be used here instead?

Reply:
I didnt write this line of code but just added the isExpanded only. Not sure if we want to change this now. I am curious about the difference between filter and find though

Reply:
`find` will give you one item that matches the condition. This is what we need to use if there is only one unique item that we are searching for.

Reply:
agreed with Imran to leave it with `filter` since the code is working fine as it is and there wont be much difference in using find instead of filter.